app/models/pipeline_run_stage.rb

```diff
@@ -49,12 +59,35 @@ def started?
     job_command.present?
   end
 
-  def failed?
-    job_status == STATUS_FAILED
+  def stage_status_file(status)
+    basename = "#{job_id}.#{status}"
+    s3_folder = case name
+                when HOST_FILTERING_STAGE_NAME
+                  pipeline_run.host_filter_output_s3_path
+                when ALIGNMENT_STAGE_NAME
+                  pipeline_run.alignment_output_s3_path
+                when POSTPROCESS_STAGE_NAME
+                  pipeline_run.postprocess_output_s3_path
+                end
+    "#{s3_folder}/#{basename}"
+  end
+
+  def assess_and_update(status_file_suffix, job_status_value)
+    # Return whether stage exited exited with status_file_suffix
+    # AND, if so, record the corresponding job_status_value in the database if it is not there already
+    answer = pipeline_run.file_generated_since_run(stage_status_file(status_file_suffix))
+    if answer && job_status != job_status_value
+      update(job_status: job_status_value)
+    end
+    answer
   end
 
-  def succeeded? # The whole thing completed successfully
-    db_load_status == 1
+  def succeeded?
+    assess_and_update(JOB_SUCCEEDED, STATUS_SUCCEEDED)
```

Comment:
it's not clear here that JOB_SUCCEEDED is a file suffix

Reply:
ok, renaming to JOB_SUCCEEDED_FILE_SUFFIX